src/potions.ts

```diff
@@ -0,0 +1,211 @@
+import {
+  cliExecute,
+  effectModifier,
+  haveEffect,
+  historicalAge,
+  historicalPrice,
+  itemAmount,
+  itemType,
+  mallPrice,
+  myAdventures,
+  numericModifier,
+  print,
+  use,
+} from "kolmafia";
+import { $effect, $familiar, $item, $items, get, have } from "libram";
+import { acquire } from "./acquire";
+import { embezzlerCount } from "./fights";
+import { baseMeat } from "./mood";
+
+const banned = $items`Uncle Greenspan's Bathroom Finance Guide`;
+
+class Potion {
+  potion: Item;
+
+  constructor(potion: Item) {
+    this.potion = potion;
+  }
+
+  effect() {
+    return effectModifier(this.potion, "Effect");
+  }
+
+  effectDuration() {
+    return numericModifier(this.potion, "Effect Duration");
+  }
+
+  meatDrop() {
+    return numericModifier(this.effect(), "Meat Drop");
+  }
+
+  familiarWeight() {
+    return numericModifier(this.effect(), "Familiar Weight");
+  }
+
+  bonusMeat() {
+    const familiarMultiplier = have($familiar`Robortender`)
+      ? 2
+      : have($familiar`Hobo Monkey`)
+      ? 1.25
+      : 1;
+
+    // Assume base weight of 100 pounds. This is off but close enough.
+    const assumedBaseWeight = 100;
+    // Marginal value of familiar weight in % meat drop.
+    const marginalValue =
+      2 * familiarMultiplier +
+      Math.sqrt(220 * familiarMultiplier) / (2 * Math.sqrt(assumedBaseWeight));
+
+    return this.familiarWeight() * marginalValue + this.meatDrop();
+  }
+
+  gross(embezzlers: number, doubleDuration = false) {
+    const bonusMeat = this.bonusMeat();
+    const duration = this.effectDuration() * (doubleDuration ? 2 : 1);
+    // Number of embezzlers this will actually be in effect for.
+    const embezzlersApplied = Math.max(
+      Math.min(duration, embezzlers) - haveEffect(this.effect()),
+      0
+    );
+
+    return (bonusMeat / 100) * (baseMeat * duration + 750 * embezzlersApplied);
+  }
+
+  price(historical: boolean) {
+    // If asked for historical, and age < 14 days, use historical.
+    return historical && historicalAge(this.potion) < 14
+      ? historicalPrice(this.potion)
+      : mallPrice(this.potion);
+  }
+
+  net(embezzlers: number, doubleDuration = false, historical = false) {
+    return this.gross(embezzlers, doubleDuration) - this.price(historical);
+  }
+
+  doublingValue(embezzlers: number, historical = false) {
+    return (
+      Math.max(this.net(embezzlers, true, historical), 0) -
+      Math.max(this.net(embezzlers, false, historical), 0)
+    );
+  }
+
+  useAsValuable(embezzlers: number, doubleDuration = false) {
+    const duration = this.effectDuration() * (doubleDuration ? 2 : 1);
```

Comment:
Should we consider the value of PYEC extension? LTC extension?

Reply:
Later I think